iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/ShuntXml.java

```diff
@@ -34,13 +40,30 @@ protected boolean hasSubElements(ShuntCompensator sc) {
         return sc != sc.getRegulatingTerminal().getConnectable();
     }
 
+    @Override
+    protected boolean hasSubElements(ShuntCompensator sc, NetworkXmlWriterContext context) {
+        return context.getVersion().compareTo(IidmXmlVersion.V_1_2) >= 0 || hasSubElements(sc);
+    }
+
     @Override
     protected void writeRootElementAttributes(ShuntCompensator sc, VoltageLevel vl, NetworkXmlWriterContext context) throws XMLStreamException {
-        if (ShuntCompensatorModelType.NON_LINEAR.equals(sc.getModelType())) {
-            throw new PowsyblException("Non linear shunt not yet supported");
-        }
-        XmlUtil.writeDouble("bPerSection", sc.getModel(ShuntCompensatorLinearModel.class).getbPerSection(), context.getWriter());
-        context.getWriter().writeAttribute("maximumSectionCount", Integer.toString(sc.getMaximumSectionCount()));
+        IidmXmlUtil.assertMinimumVersionIfNotDefault(ShuntCompensatorModelType.NON_LINEAR.equals(sc.getModelType()), ROOT_ELEMENT_NAME,
+                NON_LINEAR_MODEL, IidmXmlUtil.ErrorMessage.NOT_SUPPORTED, IidmXmlVersion.V_1_2, context);
+        IidmXmlUtil.runUntilMaximumVersion(IidmXmlVersion.V_1_1, context, () -> {
+            try {
+                XmlUtil.writeDouble(B_PER_SECTION, sc.getModel(ShuntCompensatorLinearModel.class).getbPerSection(), context.getWriter());
+                context.getWriter().writeAttribute(MAXIMUM_SECTION_COUNT, Integer.toString(sc.getMaximumSectionCount()));
+            } catch (XMLStreamException e) {
+                throw new UncheckedXmlStreamException(e);
+            }
+        });
+        IidmXmlUtil.runFromMinimumVersion(IidmXmlVersion.V_1_2, context, () -> {
+            try {
+                context.getWriter().writeAttribute(MODEL, sc.getModelType().name());
```

Comment:
I think I wouldn't serialize the type: it's a redundant information with the model you will have to check. As you do not use the same XML element name, it's not necessary.

Reply:
I changed `linearModel` and `nonLinearModel` rather than this attribute (it simplifies the file)